File: VB/WindowsApplication3/Custom LayoutControl/CanHideItemEventArgs.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Windows.Forms
Imports DevExpress.Skins

Namespace DXSample
	Public Delegate Sub CanHideItemEventHandler(ByVal sender As Object, ByVal e As CanHideItemEventArgs)

	Public Class CanHideItemEventArgs
		Inherits EventArgs
        Private canHideItem_ As Boolean = True

        Public Sub New(ByVal canHideItem_ As Boolean)
            MyBase.New()
            Me.canHideItem_ = canHideItem_
        End Sub

        Public Property CanHideLayoutItem() As Boolean
            Get
                Return canHideItem_
            End Get
            Set(ByVal value As Boolean)
                If canHideItem_ <> value Then
                    canHideItem_ = value
                End If
            End Set
        End Property
	End Class
End Namespace

File: CS/WindowsApplication3/Custom LayoutControl/CanHideItemEventArgs.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.Skins;

namespace DXSample {
    public delegate void CanHideItemEventHandler(object sender, CanHideItemEventArgs e);

    public class CanHideItemEventArgs : EventArgs
    {
        bool canHideItem = true;

        public CanHideItemEventArgs(bool canHideItem)
            : base()
        {
            this.canHideItem = canHideItem;
        }

        public bool CanHideItem
        {
            get { return canHideItem; }
            set
            {
                if (canHideItem != value)
                    canHideItem = value;
            }
        }
    }
}